misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathExpectationsSpec.kt

```diff
@@ -948,6 +950,54 @@ abstract class PathExpectationsSpec(
         }
     }
 
+    describeFun(isEmptyDirectory) {
+        val isEmptyDirectoryFun = isEmptyDirectory.lambda
+        val expectedMessage = "$isDescr: ${A_DIRECTORY.getDefault()}"
+        val expectedEmptyMessage = "$isDescr: ${AN_EMPTY_DIRECTORY.getDefault()}"
+
+        context("not accessible") {
+            it("throws an AssertionError for a non-existent path") withAndWithoutSymlink { maybeLink ->
+                val file = maybeLink.create(tempFolder.tmpDir.resolve("nonExistent"))
+                expect {
+                    expect(file).isEmptyDirectoryFun()
+                }.toThrow<AssertionError>().message {
+                    contains(expectedMessage, FAILURE_DUE_TO_NO_SUCH_FILE.getDefault())
+                    containsExplanationFor(maybeLink)
+                }
+            }
+
+            itPrintsFileAccessProblemDetails { testFile ->
+                expect(testFile).isEmptyDirectoryFun()
+            }
+        }
+
+        it("throws an AssertionError for a file") withAndWithoutSymlink { maybeLink ->
+            val file = maybeLink.create(tempFolder.newFile("test"))
+            expect {
+                expect(file).isEmptyDirectoryFun()
+            }.toThrow<AssertionError>().message {
+                contains(expectedMessage, "${WAS.getDefault()}: ${A_FILE.getDefault()}")
+                containsExplanationFor(maybeLink)
+            }
+        }
+
+        it("throws an AssertionError for a non-empty directory") withAndWithoutSymlink { maybeLink ->
+            val folder = maybeLink.create(tempFolder.newDirectory("notEmpty"))
+            maybeLink.create(tempFolder.newFile("notEmpty/a"))
+            expect {
+                expect(folder).isEmptyDirectoryFun()
+            }.toThrow<AssertionError>().message {
+                contains(expectedEmptyMessage, "${HAS.getDefault()}:")
+                contains("notEmpty/a")
+            }
+        }
```

Comment:
Can you please add a test case where the directory only contains another (empty) directory?

Reply:
The `toThrow` block should also contain the `containsExplanationFor(maybeLink)` part.

Reply:
I imagine if the directory contains another empty directory that is still not an empty directory correct (assertion will raise)?
If I include containsExplanationFor(maybeLink) the test fails, I'm not 100% sure why but I'm thinking it has to do with  how Files.list work. isEmptyDirectory does follow symbolic links.

Reply:
> I imagine if the directory contains another empty directory that is still not an empty directory correct (assertion will raise)?

Yes, the assertion should fail for a directory that contains another empty directory.

> If I include containsExplanationFor(maybeLink) the test fails

Without seeing the error message, I can’t tell why the test fails. The error message should give you hints. However, I _suspect_ that your assertion does not report symlinks that have been resolved along the way. There is the helper function `explainForResolvedLink` (defined in `ch.tutteli.atrium.logic.creating.filesystem.hints.hits.kt`) which provides the functionality for explaining the symbolic link chain. Please use it for your assertion, as well. You can look at the other assertions to see how it should be used.